### ISSUE TITLE
Either support converting to converter arg types or explicitly disallow non-JSON argument types

### ISSUE THREAD

**Issue by kevmoo** [google/json_serializable.dart#806]
See https://github.com/google/json_serializable.dart/issues/795

Converter takes type of `List<String>`.

generates

```dart
LoginState _$LoginStateFromJson(Map<String, dynamic> json) {
  return LoginState(
    cookies:
        const _CookiesJsonConverter().fromJson(json['cookies'] as List<String>),
  );
}
```

Of course the `as List<String>` will fail always.

We could/should do better here!

Annoying, but this is not a blocker. User can just change the converter to do the right thing.

**Comment by kevmoo** [comment]
It's a bit gnarly. I'll have to dig in there!

sadly, I'm blocked by another issue - https://github.com/dart-lang/sdk/issues/45127
